src/flightlog.js

```diff
@@ -726,33 +722,49 @@ export function FlightLog(logData) {
 
       if (!destChunk.hasAdditionalFields) {
         destChunk.hasAdditionalFields = true;
-
-        let chunkIMU = new IMU(sourceChunks[sourceChunkIndex].initialIMU);
+        const chunkIMU = new IMU(sourceChunk.initialIMU);
 
         for (let i = 0; i < sourceChunk.frames.length; i++) {
           let srcFrame = sourceChunk.frames[i],
             destFrame = destChunk.frames[i],
             fieldIndex = destFrame.length - ADDITIONAL_COMPUTED_FIELD_COUNT;
 
-          if (!that.isFieldDisabled().GYRO) {
-            //don't calculate attitude if no gyro data
-            attitude = chunkIMU.updateEstimatedAttitude(
-              [
-                srcFrame[gyroADC[0]],
-                srcFrame[gyroADC[1]],
-                srcFrame[gyroADC[2]],
-              ],
-              [
-                srcFrame[accSmooth[0]],
-                srcFrame[accSmooth[1]],
-                srcFrame[accSmooth[2]],
-              ],
-              srcFrame[FlightLogParser.prototype.FLIGHT_LOG_FIELD_INDEX_TIME],
-              sysConfig.acc_1G,
-              sysConfig.gyroScale,
-              magADC
-            );
+          if (imuQuaternion) {
+            const scaleFromFixedInt16 = 0x7FFF; // 0x7FFF = 2^15 - 1
+            const q = {
+              x: srcFrame[imuQuaternion[0]] / scaleFromFixedInt16,
+              y: srcFrame[imuQuaternion[1]] / scaleFromFixedInt16,
+              z: srcFrame[imuQuaternion[2]] / scaleFromFixedInt16,
+              w: 1.0,
+            };
+            q.w = Math.sqrt(1.0 - (q.x ** 2 + q.y ** 2 + q.z ** 2));
```

Comment:
@demvlad think @ledvinap was hinting at
```suggestion
              w: Math.sqrt(1.0 - (q.x ** 2 + q.y ** 2 + q.z ** 2));
            };
```